sdks/python/src/opik/validation/chat_prompt_messages.py

```diff
@@ -0,0 +1,120 @@
+
+            # Validate message has exactly "role" and "content" keys
+            message_keys = set(message.keys())
+            expected_keys = {"role", "content"}
+
+            if message_keys != expected_keys:
+                if not message_keys.issubset(expected_keys):
+                    missing_keys = expected_keys - message_keys
+                    msg = f"{prefix}: missing required keys: {sorted(missing_keys)}"
+                    failure_reasons.append(msg)
+                if not expected_keys.issubset(message_keys):
+                    extra_keys = message_keys - expected_keys
+                    msg = (
+                        f"{prefix}: unexpected keys: {sorted(extra_keys)}. "
+                        f"Expected only: {sorted(expected_keys)}"
+                    )
+                    failure_reasons.append(msg)
+                continue
+
+            # Validate role
+            role = message.get("role")
+            if role not in self.VALID_ROLES:
+                valid_roles_str = ", ".join(
+                    [f"'{r}'" for r in sorted(self.VALID_ROLES)]
+                )
+                msg = (
+                    f"{prefix}.role: must be one of [{valid_roles_str}] "
+                    f"but {repr(role)} was given"
+                )
+                failure_reasons.append(msg)
+
+            # Validate content
+            content = message.get("content")
+            if content is None:
+                msg = f"{prefix}.content: must not be None"
+                failure_reasons.append(msg)
+            elif not isinstance(content, (str, list)):
+                msg = (
+                    f"{prefix}.content: must be either str or list of dicts "
+                    f"but {type(content).__name__} was given"
+                )
+                failure_reasons.append(msg)
+            elif isinstance(content, list):
+                # Validate content is list of dicts with "type" key
+                for content_idx, content_part in enumerate(content):
+                    content_prefix = f"{prefix}.content[{content_idx}]"
+                    if not isinstance(content_part, dict):
+                        msg = (
+                            f"{content_prefix}: must be a dict "
+                            f"but {type(content_part).__name__} was given"
+                        )
+                        failure_reasons.append(msg)
+                    elif "type" not in content_part:
+                        msg = f"{content_prefix}: must have 'type' key"
+                        failure_reasons.append(msg)
+
+        # Create validation result
+        if len(failure_reasons) > 0:
+            self.validation_result = result.ValidationResult(
+                failed=True, failure_reasons=failure_reasons
+            )
+        else:
+            self.validation_result = result.ValidationResult(failed=False)
+
+        return self.validation_result
+
+    def raise_validation_error(self) -> None:
```

Comment:
Maybe `raise_if_failed()` is a more suitable name. Without that it's a bit confusing to see `raise_validation_error()` call in the ChatPrompt class code without any conditions.